api/db_queries.py

```diff
@@ -66,4 +66,13 @@
     SORT category.categorynr
     RETURN {category: category.category,
             categoryname: CONCAT_SEPARATOR(" ",UPPER(category.category),category.categoryname)}
+"""
+
+query_graph_data = """
+FOR p IN parallels
+    FILTER p._key LIKE CONCAT(@filename,"%")
+    FILTER p.score >= @score
+    FILTER p.par_length >= @parlength
+    FILTER p["co-occ"] <= @coocc
+    RETURN SPLIT(p.par_segnr[0],":")[0]
```

Comment:
unfortunately this is a little too slow at the moment. To avoid using `LIKE CONCAT`, maybe when loading the data, we could add a `filename` property to the `parallel` object and filter by that?

Reply:
Would this be better: `FILTER SPLIT(p._key,":")[0] == @filename`

Reply:
I think adding an extra property is a bit strange if the property is already in the key.

Reply:
the property is in the key, but calling `LIKE` and `CONCAT` slows down the speed of the query by a lot. Currently this query takes 2-6 seconds for me, when it should be around 30ms, maximum 300ms. You can only access data rows that quickly using indexed columns, not dynamically computed parameters.

But I agree adding the property might not be the best solution. Maybe a better one would be to create an extra table (document) like `files-parallels` that would hold arrays of parallels for a given file? Or, to make it simpler, add that to the `files` collection?

Then we could just query for the filename and get a list of parallels in that file.

Reply:
By the way, the items returned by this query are not unique. We can count them in AQL instead:
```
FOR p IN parallels
    FILTER p._key LIKE CONCAT(@filename,"%")
    FILTER p.score >= @score
    FILTER p.par_length >= @parlength
    FILTER p["co-occ"] <= @coocc
    COLLECT textname = SPLIT(p.par_segnr[0],":")[0] WITH COUNT INTO length
    RETURN {
        "textname": textname,
        "length": length
    }
```
returns:
```json
[
  {
    "textname": "an9.20",
    "length": 5
  },
  {
    "textname": "dn16",
    "length": 1
  },
  {
    "textname": "dn26",
    "length": 3
  },
]
```

Reply:
This wouldn't work as you are counting matches per file, not per collection.

Reply:
Ah, I see. But I still think it's better to return this structure and use python to add the `length` numbers together per collection. Otherwise we pass duplicate values. For example, we can pass this:

```json
  {
    "textname": "dn26",
    "length": 1000
  },
```

or this: `["dn26", "dn26", "dn26", "dn26", ...]`, but in the second example python would have to iterate a 1000 times needlessly.

I may not be seeing something though, if so - forgive me!